### ISSUE TITLE
ACE by Daisy

### ISSUE THREAD

**Issue by s-lhj** [nlbdev/nordic-accessible-epub-guidelines#73]
ACE by Daisy has some major vulnerabilities:

9.8	CRITICAL	CVE-2022-39353⁠	https://scout.docker.com/vulnerabilities/id/CVE-2022-39353?s=github&n=xmldom&t=npm&vr=%3C%3D0.6.0&utm_source=desktop&utm_medium=ExternalLink
9.8	CRITICAL 	CVE-2022-37616	https://scout.docker.com/vulnerabilities/id/CVE-2022-37616?s=gitlab&n=xmldom&t=npm&vr=%3C%3D0.7.6&utm_source=desktop&utm_medium=ExternalLink
7.5	HIGH		CVE-2024-37890	https://scout.docker.com/vulnerabilities/id/CVE-2024-37890?s=github&n=ws&t=npm&vr=%3E%3D8.0.0%2C%3C8.17.1&utm_source=desktop&utm_medium=ExternalLink
7.5	HIGH		CVE-2024-45296⁠	https://scout.docker.com/vulnerabilities/id/CVE-2024-45296?s=github&n=path-to-regexp&t=npm&vr=%3C0.1.10&utm_source=desktop&utm_medium=ExternalLink
7.5	HIGH		CVE-2022-24434	https://scout.docker.com/vulnerabilities/id/CVE-2022-24434?s=github&n=dicer&t=npm&vr=%3C%3D0.3.1&utm_source=desktop&utm_medium=ExternalLink
...

and these are the gravest vulnerabilities in our system at Statped. The vulnerability level has been stable for at least the last two years (from the time we started checking), although the type of vulnerabilities change over time. We have discussed cutting ACE all together, but in this context we might consider reevaluating paragraph 2.11.

@AndersEkl , @TorilBWM , @indixcss

**Comment by oscarlcarlsson** [comment]
Can you please elaborate on the issue here? Does this impact the requirements from the suppliers?

**Comment by s-lhj** [comment]
This is from a scan of the Docker image, but I thing other scans of the ACE project would yield similar results. Earlier (2023), the biggest problems were found in other dependencies, and these were also critical. I suggest you do a scan both with Docker modules and in GitHub, for example with Snyk.

Since our suppliers know that a) we have systems with these vulnerabilities, and b) that they send files that are to be sent directly into systems with these vulnerabilities, I think this is a security issue. At Statped, we have started the process of dockerizing everything, and we have discussed the possibility of cutting out ACE from the system. 

I tried to mention this to other actors like Dedicon, but I have no contacts in Daisy. Surely they must have done a scan of their own code, so it is not like they do not know of the problems.

For out part, I just think it is slightly problematic to require the inclusion of vulnerabilities ranked as critical. There are workarounds, like encapsulating systems, but still.

Other questions?

**Comment by s-lhj** [comment]
Looks like "[josteinaj](https://github.com/nlbdev/nordic-epub3-dtbook-migrator/commits?author=josteinaj)
[comment out Ace references](https://github.com/nlbdev/nordic-epub3-dtbook-migrator/commit/fdc104890587a07851c54bb231dc4967bd6a4c3d)" two years ago in the validator.

**Comment by oscarlcarlsson** [comment]
Do you see an issue with the producers using ACE for validations on their end? We don't have to use it on our end if we don't want to. But it's still a useful tool for us to check that our guidelines comply with general accessibility standards.

**Comment by s-lhj** [comment]
From their perspective it might seem harsh to be obliged to use vulnerable software.

**Comment by s-lhj** [comment]
No, I haven't, but then again, they should do these checks themselves.

As for formulating the passage, "should" for epubcheck and "could" for ace could be a way.

**Comment by jonaslil** [comment]
Getting back to this issue. Here's my proposal, which clarifies that using Ace is optional rather than required.

Current text:

> Validation of EPUB files should be done using [EPUBCheck](https://www.w3.org/publishing/epubcheck/) for structural validity and conformance to the EPUB standard, and the [Ace by DAISY](https://daisy.github.io/ace/) tool for accessibility evaluation.

Proposed edit:

> Validation of EPUB files should be done using EPUBCheck for structural validity and conformance to the EPUB standard. Additionally, Ace by DAISY may be used for accessibility evaluation.

**Comment by s-lhj** [comment]
Fine by me.
